tools/pipelines/templates/include-set-package-version.yml

```
@@ -101,17 +108,41 @@ steps:
       echo VERSION_RELEASE=$VERSION_RELEASE
       echo VERSION_PATCH=$VERSION_PATCH
       echo VERSION_INCLUDE_INTERNAL_VERSIONS=$VERSION_INCLUDE_INTERNAL_VERSIONS
+      echo PACKAGE_TYPES_FIELD=$PACKAGE_TYPES_FIELD
 
       # Generate the build version. Sets the environment variables version, codeVersion, and isLatest.
       # These are referenced in following steps prefixed by this task name. E.g. SetVersion.version
       flub generate buildVersion
 
+- ${{ if ne(parameters.packageTypesOverride, 'none') }}:
+  - task: Bash@3
+    displayName: Set Package Types - ${{ parameters.packageTypesOverride }}
+    inputs:
+      targetType: 'inline'
+      workingDirectory: ${{ parameters.buildDirectory }}
+      script: |
+        flub release setPackageTypesField -g ${{ parameters.tagName }} --types ${{ parameters.packageTypesOverride }}
+
+# Set INTERDEPENDENCY_RANGE
+- task: Bash@3
+  name: calculateInterdependencyRange
+  displayName: Calculate INTERDEPENDENCY_RANGE
+  inputs:
+    targetType: 'inline'
+    workingDirectory: ${{ parameters.buildDirectory }}
+    script: |
+      if [[ "${{ parameters.packageTypesOverride }}" == "alpha" ]] || [[ "${{ parameters.packageTypesOverride }}" == "beta" ]]; then
+        echo "##vso[task.setvariable variable=INTERDEPENDENCY_RANGE]exact"
+      else
+        echo "##vso[task.setvariable variable=INTERDEPENDENCY_RANGE;isOutput=true]$(interdependencyRange)"
+      fi
+
 - task: Bash@3
   displayName: Update Package Version (flub)
   env:
     VERSION_RELEASE: $(release)
     RELEASE_GROUP: ${{ parameters.tagName }}
-    INTERDEPENDENCY_RANGE: ${{ parameters.interdependencyRange }}
+    INTERDEPENDENCY_RANGE: $(calculateInterdependencyRange.INTERDEPENDENCY_RANGE)
```

```suggestion
    ${{ if or(eq(parameters.packageTypesOverride, 'alpha'), eq(parameters.packageTypesOverride, 'beta')) }}:
      INTERDEPENDENCY_RANGE: exact
    ${{ else }}:
      INTERDEPENDENCY_RANGE: ${{ parameters.interdependencyRange }}
```

This is another alternative.

---

we do not need to set `INTERDEPENDENCY_RANGE` here. I was going through this script: https://github.com/microsoft/FluidFramework/blob/main/scripts/update-package-version.sh#L20 and if it is a non-release build, then `exact` interdependencies are always used.